snuba/datasets/storage.py

```diff
@@ -157,6 +158,45 @@ def get_mandatory_condition_checkers(self) -> Sequence[ConditionChecker]:
         return self.__mandatory_condition_checkers
 
 
+class PartitionedReadableTableStorage(ReadableStorage):
```

Comment:
I'm not sure I want us to introduce this class, I see that it's for safety so that `get_cluster()` can't be used improperly, but IMO we should just add the `partition_id: Optional[int] = None` parameter to the parent class and ignore it almost all the time